lib/ansible/modules/system/zfs_permissions.py

```diff
@@ -0,0 +1,271 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2015, Nate Coraor <nate@coraor.org>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+DOCUMENTATION = '''
+---
+module: zfs_permissions
+short_description: Manage zfs administrative permissions
+description:
+  - Manages ZFS file system administrative permissions on Solaris and FreeBSD. See zfs(1M) for more information about the properties.
+version_added: "2.2"
+options:
+  name:
+    description:
+      - File system or volume name e.g. C(rpool/myfs)
+    required: true
+  state:
+    description:
+      - Whether to allow (C(present)), or unallow (C(absent)) a permission.
+    required: true
+    choices: [present, absent]
+  users:
+    description:
+      - Users to whom permission(s) should be granted, separated by commas.
+    required: false
+  groups:
+    description:
+      - Groups to whom permission(s) should be granted, separated by commas.
+    required: false
+  everyone:
+    description:
+      - Apply permissions to everyone.
+    required: false
+    default: false
+    choices: ['on','off']
```

Comment:
You are saying that on/off is the accepted values, yet in argument_spec you are using BOOLEANS, which you are converting with module.boolean. Why not accepting booleans here and remove the call to module.boolean()? Also defining type: bool for everyone/local/descendents/recursive would be more consistent with other modules.